### PR TITLE
Chart service misc

### PR DESCRIPTION
**Pull request by rnugraha**

### CODE REVIEW

File: src/app/components/cohort-charts/cohort-chart.directive.js [hunk 1]
```diff
@@ -43,6 +43,7 @@ angular.module('transmartBaseUi')
                     'tsGridsterItem.sizeX', 'tsGridsterItem.sizeY',
                     'tsGridster.curColWidth', 'tsGridster.curRowHeight'
                 ], function (newValues, oldValues, scope) {
+                    console.log(_cached);
```
Comment (berndvdveen) [inline src/app/components/cohort-charts/cohort-chart.directive.js]: :fish:

File: src/app/components/cohort-charts/chart.service.js [hunk 2]
```diff
@@ -138,27 +117,33 @@ angular.module('transmartBaseUi').factory('ChartService',
 
             /**
              * Add new label to list and check data type
-             * @param label
-             * @param value
+             * @param obs
+             * @param node
+             * @param filterObj
+             * @returns {*}
```
Comment (berndvdveen) [inline src/app/components/cohort-charts/chart.service.js]: Returns label ID right?

File: src/app/components/cohort-charts/chart.service.js [hunk 3]
```diff
@@ -138,27 +117,33 @@ angular.module('transmartBaseUi').factory('ChartService',
...
+             * @private
              */
             var _addLabel = function (obs, node, filterObj) {
 
                 // Check if label has already been added
                 var label = _.find(chartService.cs.labels, {label: obs.label});
                 var filters;
-                if (filterObj) filters = filterObj.filterWords;
+
+                if (filterObj) {
```
Comment (berndvdveen) [inline src/app/components/cohort-charts/chart.service.js]: Clarify conditional

File: src/app/components/cohort-charts/chart.service.js [hunk 4]
```diff
@@ -231,32 +194,26 @@ angular.module('transmartBaseUi').factory('ChartService',
...
 
                             // Check if the subject of the observation is already present
-                            var found = _.find(chartService.cs.subjects, {id: obs._embedded.subject.id});
+                            var foundSubject = _.find(chartService.cs.subjects, {id: obs._embedded.subject.id});
 
-                            if (found) {
-                                found.labels[_id] = obs.value;
+                            if (foundSubject) {
+                                foundSubject.labels[_newLabelId] = obs.value;
+                                console.log('subject found ..', foundSubject)
```
Comment (berndvdveen) [inline src/app/components/cohort-charts/chart.service.js]: Remove logging

File: src/app/components/cohort-charts/chart.service.js [hunk 5]
```diff
@@ -306,27 +263,26 @@ angular.module('transmartBaseUi').factory('ChartService',
...
+                    return chartToBeRemoved.id !== label.labelId;
                 });
             };
 
             /**
              * Clear chart's filter
              * @memberof ChartService
-             * @param  {Array}charts
              * @param {Object} label
              * @returns {*}
```
Comment (berndvdveen) [inline src/app/components/cohort-charts/chart.service.js]: Returns chart obj

File: src/app/components/cohort-charts/chart.service.js (hunk 3 above)
Comment (rnugraha) [inline src/app/components/cohort-charts/chart.service.js]: this i assume will check the truthy value of filterObj

**Comment by kretep** [comment]
Looks ok

**Pull request merged**